src/arch/arm/64/kernel/vspace.c

```diff
@@ -943,7 +943,7 @@ exception_t handleVMFault(tcb_t *thread, vm_fault_type_t vm_faultType)
 #ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
         /* use the IPA */
         if (ARCH_NODE_STATE(armHSVCPUActive)) {
-            addr = GET_PAR_ADDR(ats1e1r(addr)) | (addr & MASK(PAGE_BITS));
+            addr = GET_PAR_ADDR(addressTranslateS1(addr)) | (addr & MASK(PAGE_BITS));
```

Comment:
The aarch64 code returns IPA previously, this is not the VA you mentioned in the commit?

Reply:
Well spotted. Sigh. I intended to refer to `Arch_setMRs_fault` in `src/arch/arm/api/faults.c`.

Reply:
On AArch64, that would not have given you an IPA before this commit, but my commit message is somewhat confused. @kent-mcleod : help?